### PR TITLE
Geofencing sample: build upgrades

### PR DESCRIPTION
**Pull request by jdkoren**
- Bump versions of Gradle and AGP
- Minor fixes to make `gradlew lint test build` succeed

Change-Id: I6eebf318bfe8a04c721175a7c8ba9a0179c08c61

### CODE REVIEW

File: Geofencing/app/build.gradle [hunk 1]
```diff
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 28
+    compileSdkVersion 31
```
Comment (javadude) [inline Geofencing/app/build.gradle:4]: why not 33?

**Comment by javadude** [comment]
Also - IIRC there were several things in the code that needed to change, possibly for updated other dependencies. Not sure what they were, but I know a full upgrade was pretty painful. Might have been wrt updating permissions to the registerActivityResult form?

File: Geofencing/app/build.gradle (hunk 1 above)
Comment (jdkoren) [reply Geofencing/app/build.gradle]: Done

**Comment by jdkoren** [comment]
> Also - IIRC there were several things in the code that needed to change, possibly for updated other dependencies. Not sure what they were, but I know a full upgrade was pretty painful. Might have been wrt updating permissions to the registerActivityResult form?

Yes, there are definitely a lot of things which need fixing. Doing all of that at once would make for a very large PR, so I just tried to do the minimum that would build with the updated Gradle and AGP versions.

**Comment by javadude** [comment]
Was just testing it out and got

```
Caused by: java.lang.IllegalArgumentException: com.google.android.gms.location.sample.geofencing: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
```

when trying to add a geofence. Looks familiar. I started to go down the route of making changes (adding PendingIntent.FLAG_IMMUTABLE and updating the min SDK to 23 and updating dependencies) and it keeps saying

```
W/MainActivity: Unknown error: the Geofence service is not available now
```

Did it run for you? I'm running on a Pixel 7 with Android 13

**Comment by jdkoren** [comment]
My goal was just for it to build. This sample is pretty stale and I don't expect it to run properly without several changes.

**Comment by javadude** [comment]
Ok - I just verified that the existing app main branch builds but has the same error when running. Should we just remove it?

**Review by javadude — APPROVED** [review]
Looks good - the Geofencing sample was already broken and the build changes don't worsen it.
We're looking at a sample deprecation process to clean this up as well as other samples that are outdated or no longer work.

**Pull request closed without merging**